### PR TITLE
#3169 program cr updates

### LINKED ISSUE #3169: ProgramCustomer requistion updates
## Describe the bug

From initial feedback on program based customer requisitions there are a few minor improvements to the UI.

### Proposed Solution

- The period & comment page info fields should be switched around.
- The months of stock should use maxMOS as the wording.
- There is a bug with focusing the next row.
- The column widths could use some adjusting.
- The their column has 19 different names

### Version and device info

- App version: 6.0
- Tablet model: N/A
- OS version: N/A

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #3169 

## Change summary

- Switched page info around to make a bit more sense
- Changed some localization
- Added the formulas for number of days in a month and adjusted formulas to make a bit more sense
- Fixed a bug with focusing the next cell
- Adjusted column widths slightly

## Testing

N/A

### Related areas to think about

N/A

### CODE REVIEW

File: src/database/DataTypes/Requisition.js [hunk 1]
```diff
@@ -130,7 +130,7 @@ export class Requisition extends Realm.Object {
    * @return  {number}
    */
   get monthsToSupply() {
-    return this.daysToSupply / NUMBER_OF_DAYS_IN_A_MONTH;
+    return Math.ceil(this.daysToSupply / NUMBER_OF_DAYS_IN_A_MONTH);
```
Comment (joshxg) [inline src/database/DataTypes/Requisition.js:133]: This is used for display

File: src/localization/pageInfoStrings.json [hunk 2]
```diff
@@ -16,17 +16,20 @@
...
-    "suggested_formula": "DAILY CONSUMPTION x DAYS TO SUPPLY - STOCK ON HAND",
-    "amc_equals": "Average monthly consumption",
-    "amc_formula": "OUTGOING STOCK x NUMBER OF DAYS IN A MONTH / (DAYS IN PERIOD - DAYS OUT OF STOCK)",
-    "none_suggested": "None suggested"
+    "suggested_equals": "Suggested quantity",
+    "suggested_formula": "(Monthly use x Max MOS) - Their stock",
+    "amc_equals": "Monthly use",
+    "number_of_days_in_a_month": "Number of days in a month",
+    "number_of_days_in_a_month_formula": "= Number of days in a year / Number of months in a year",
+    "amc_formula": "(OUT x Number of days in a month) / (Days within period - DOS)",
```
Comment (joshxg) [inline src/localization/pageInfoStrings.json:30]: Adjusted these formulas slightly to refer back to the column names and added parentheseseis cause Rowena said its better with them

File: src/localization/programStrings.json [hunk 3]
```diff
@@ -12,7 +12,7 @@
     "no_periods": "No periods available",
     "program": "Program",
     "supplier": "Supplier",
-    "order_type": "Order type",
+    "order_type": "Order Type",
```
Comment (joshxg) [inline src/localization/programStrings.json:15]: Apparently this is correct english

File: src/pages/dataTableUtilities/getColumns.js [hunk 4]
```diff
@@ -48,7 +48,7 @@ const PAGE_COLUMN_WIDTHS = {
   sensors: [3, 3, 1, 1],
   [TABS.ITEM]: [1, 3, 1],
   [TABS.PRESCRIBER]: [3, 3, 1],
-  editableCustomerRequisitionFormEntry: [2, 2, 1, 1, 1, 1, 2, 2, 2, 2],
+  editableCustomerRequisitionFormEntry: [2, 1.5, 1, 1, 1, 1.5, 1, 1, 1, 1.5],
```
Comment (joshxg) [inline src/pages/dataTableUtilities/getColumns.js:51]: Slight column width adjustments

File: src/pages/dataTableUtilities/getColumns.js [hunk 5]
```diff
@@ -641,10 +641,10 @@ const COLUMNS = () => ({
   [COLUMN_NAMES.CLOSING_STOCK]: {
     type: COLUMN_TYPES.NUMERIC,
     key: COLUMN_KEYS.CLOSING_STOCK,
-    title: tableStrings.closing_stock,
+    title: tableStrings.their_stock,
```
Comment (joshxg) [inline src/pages/dataTableUtilities/getColumns.js:644]: Consistent naming

File: src/pages/dataTableUtilities/getColumns.js [hunk 6]
```diff
@@ -641,10 +641,10 @@ const COLUMNS = () => ({
...
     alignText: 'right',
     sortable: true,
-    editable: true,
+    editable: false,
```
Comment (joshxg) [inline src/pages/dataTableUtilities/getColumns.js:647]: This fixes the focusing of the next cell bug

File: src/pages/dataTableUtilities/getPageInfoColumns.js [hunk 7]
```diff
@@ -65,7 +65,7 @@ const PER_PAGE_INFO_COLUMNS = {
   ],
   [ROUTES.CUSTOMER_REQUISITIONS_WITH_PROGRAMS]: [
     ['program', 'orderType', 'monthsToSupply', 'entryDate'],
-    ['customer', 'requisitionComment', 'period', 'createdDate'],
+    ['period', 'customer', 'createdDate', 'requisitionComment'],
```
Comment (joshxg) [inline src/pages/dataTableUtilities/getPageInfoColumns.js:68]: Adjusted these slightly as apparently it makes more sense said Rowena!

File: src/pages/dataTableUtilities/getPageInfoColumns.js [hunk 8]
```diff
@@ -86,8 +86,11 @@ const PER_PAGE_INFO_COLUMNS = {
     ['breachDuration', 'location'],
     ['numberOfAffectedBatches', 'affectedQuantity'],
   ],
+
   customerRequisitionItemDetail: [
     [
+      'numberOfDaysInAMonthString',
+      'numberOfDaysInAMonthFormula',
```
Comment (joshxg) [inline src/pages/dataTableUtilities/getPageInfoColumns.js:93]: It was confusing refering to the number of days in a month without specifying what that was

File: src/widgets/CustomerRequisitionItemDetail.js [hunk 9]
```diff
@@ -17,9 +17,11 @@ import { DARKER_GREY, WHITE, SUSSOL_ORANGE } from '../globalStyles';
 const CustomerRequisitionItemDetailsComponent = ({ item }) => (
   <View style={localStyles.container}>
     <PageInfo
+      titleTextAlign="right"
       titleColor={SUSSOL_ORANGE}
       infoColor={WHITE}
       columns={getPageInfoColumns('customerRequisitionItemDetail')(item)}
+      numberOfLines={2}
```
Comment (joshxg) [inline src/widgets/CustomerRequisitionItemDetail.js:24]: Added number of lines and text align pass through to align the title of page info to the right when needed and multiple lines when needed. Default prop to be what the old hard coded values were.

**Review by mark-prins — APPROVED** [review]
changes look good to me!

**Comment by joshxg** [comment]
@mark-prins Thanks!!

**Pull request merged**